create folder for tasks with subtasks

johny4437 opened the pull request:
Fixe bug ( PRIORITY MEDIUM) - 5 from PROJECT/TASK/WBS COMPONENT
Description
Creating sub-tasks in a WBS doesn’t create a new folder and makes the tasks uneditable
Logged in as Admin → Other Links → Projects → AAA USE THIS PROJECT → Click WBS Icon → Choose “Jae's WBS - USE THIS” → Choose task not in folder → Edit → Add Task → Should create folder out of parent and including new task

## Related PRS (if any):
This frontend PR is related to the [#283](https://github.com/OneCommunityGlobal/HGNRest/pull/283) backend PR.


Mainly changes explained:
1 - Creating sub-tasks in a WBS create a new folder.
2 - Make subtasks editable and can be deleted.

How to test:
1 - check into current branch
2 - do `npm install` and ... to run this PR locally
3 - log as admin user
4 - Go to dashboard→ Other Links→ Projects→ chose the Project "AAA USE THIS PROJECT"→ click "[ Jae's WBS - USE THIS]"
→ Add a task → Create a subtask for this task
![Captura de Tela (71)](https://user-images.githubusercontent.com/39320057/220209311-b746c0c7-2cc0-4d75-a240-3d8bd61439b7.png)
![Captura de Tela (72)](https://user-images.githubusercontent.com/39320057/220209378-6d65bc32-91f3-4735-a961-c3432c3f6d47.png)

suigunfjc commented:
Hi @johny4437 , I tested the feature in both your branch and the dev branch. Here is the output:

In dev branch: the sub-task will be actually created and can be displayed by clicking the mother task. However, there isn't a folder shown before the mother task after refresh.

In your branch: the sub-task can be created and there is a folder at the mother task.

I think you fixed the bug, good job! I will go through the code later to see if there are any potential bugs.

https://user-images.githubusercontent.com/94319381/220258811-ecd39478-9a39-432e-adb1-c7cd78132c53.mp4

NyahJ commented:
Hi @johny4437,

I tested to create sub-tasks in a WBS create a new folder and the subtasks editable and can be deleted.
Thanks for your effort to fix this bug!

<img width="1034" alt="截屏2023-02-20 下午9 49 06" src="https://user-images.githubusercontent.com/77643813/220258589-baed191b-ae65-4a9d-84f3-8f471801672a.png">

Lunaliar commented:
Hey @johny4437,
I was able to make a task and then turn that task into a parent folder for other tasks great! I did run into a few issues. I'm not sure exactly why but when I make tasks they have the same number as another user. When this happens it leads to problems with unfolding below the lowest folder and if my folder is deleted there is a chance that the other subtask on the other user's folder may be deleted as well.
-Making a folder task (Testing of 682).
![delete other folder subtasks](https://user-images.githubusercontent.com/98275089/220443550-5ced3a9d-0b55-4d42-a795-97300386f1e0.png)
-Subtask of 682 under Jinchaof 682 instead of Testing 682.
![Make a task into a folder](https://user-images.githubusercontent.com/98275089/220443552-1f9f5d01-9498-4c05-bccd-f0e4895cdb2d.png)
-Deleting subtasks of Jinchaof 682 when Testing of 682 folder is deleted.
![subtask below a folder](https://user-images.githubusercontent.com/98275089/220443558-e32d0f40-e467-44d1-9ca9-dd09bc18e5ea.png)

I also found a few housekeeping items if you want to take care of them before we merge.
- SingleTask/SingleTask.js line 36 has a debug statement
- WBSDetail/Task/Task.jsx has a debug statement on line 154
- WBSDetail/WBSTasks.jsx lines 186-190 we can use let instead of var

agujjar2206 commented:
Hi @johny4437, 

I tested this PR using the following steps:
1. Created a main task "parent".
2. Created 2 sub tasks "child1" and "child2" inside "parent".
3. Removed "child2".
4. I found that when removing a sub-task, both the removed task and the remaining task were removed from the list.

Note: I also checked out the backed PR #283.

Thank you.

![682-children](https://user-images.githubusercontent.com/94432002/220452665-fdb47083-24e4-48ea-861b-d9da3bb948f2.png)
![682-parent](https://user-images.githubusercontent.com/94432002/220452697-0fd780c1-dbee-48c4-8211-fb6912cc8676.png)
![682-parent2](https://user-images.githubusercontent.com/94432002/220452708-3f8e0228-506f-4b7f-8888-51237d54b364.png)

kamtoeddy reviewed (commented):
Hi @johny4437,

Just tested and it works perfectly.  I created task number 56.1

The code also looks fine

![Screenshot (496)](https://user-images.githubusercontent.com/65472478/220459158-4121241a-7498-403c-a0ba-b3d7feaa9fea.png)

Lucas-E reviewed (commented):
Hey Johny, how are you doing? Hope you are doing fine. I tested both development and your branch, and everything you did seems to be working fine, great job!
![JohnnyPr682FirstReview](https://user-images.githubusercontent.com/92558269/220465676-b4034c91-4e61-4005-9237-020c13eee792.png)

tingedudu8 commented:
Hi @johny4437, I tested your PR and found out that if there are two folders with the same serial number,  the subtasks will be created under the wrong folder. Also, if the subtasks are all deleted, i guess it is better to make the parent folder back to file format.
Thanks!
https://user-images.githubusercontent.com/70128933/220465875-9b77a35a-b1ec-4d0e-a0ef-3b10a3ecb496.mov

johny4437 commented:
Hi @Lunaliar I removed both debug lines, but the archive `WBSDetail/WBSTasks.jsx lines 186-190 ` is not related to my PR I think what happened was that I ended up applying eslint in this archive and then I did not pay attention and I added this togheter with my PR.

Jorge-Nerd commented:
Hi @johny4437 I tested on development branch 
![image](https://user-images.githubusercontent.com/77848827/220479626-5e04a6cb-9ffe-4886-ae05-adc28ab63581.png)

and compared to your branch it seem to fix the bug good job
![image](https://user-images.githubusercontent.com/77848827/220479863-4e5c274b-b8f6-49e0-8fc1-3cd405d23405.png)

angelinatr15 commented:
Hello @johny4437,

I tested your branch with the backend PR #283. I think it looks great! I was able to create a task within another task which wrapped it into a folder that also allows me to edit it as well. I was also able to delete the folder and task that was created. 

When I created a another subtask under the subtask and tried to edit it the original text in "Why this Task is Important"  did not appear. Therefore I inputted a new text into that field and updated it. When I checked it again the new text "testing" did not appear into the field but the original input text appeared. This might just be a small delay because when I perform this test over again everything worked fine and the text field appeared as expected. 

https://user-images.githubusercontent.com/97488347/220521911-8a7a46b0-06ff-4548-b57c-8350452b88a7.mov

AnishPandita commented:
Hi, I tested your PR and the creation of subtask and editing them seems to work as intended. However when i created two subtasks under a task and deleted one of the subtasks the other subtask got deleted too. Here is the video for the same.

https://user-images.githubusercontent.com/111321945/220568459-867e42be-9164-4727-adb5-b314ec9d3994.mov

vanelnw commented:
Hi @johny4437 
I tested your PR and was able to  create a subtasks and edit them. All work as expected
<img width="1399" alt="Capture d’écran 2023-02-23 à 03 05 04" src="https://user-images.githubusercontent.com/38879488/220807661-438c92ba-f3cf-47e0-944e-af446833f8cb.png">

kamtoeddy commented on src/components/Projects/WBS/SingleTask/SingleTask.js:
```diff
@@ -13,6 +13,7 @@ import hasPermission from 'utils/permissions';
 
 const SingleTask = props => {
   const taskId = props.match.params.taskId;
+  console.log('TASK ID PARAM', typeof taskId);
```
Hi @johny4437,

Please can you remove line 16?

harlleybastos commented:
hi @johny4437, I just tested it here and It seems to be working.

https://user-images.githubusercontent.com/37200778/221001467-c203dd55-c12b-4e6a-9428-811bc8f8cc24.mp4

My test step by step:

1. Create the task: "harlley-task-test"
2. Create the subtask: "harlley-subtask-test"
3. It seems to be working!

vitorsaa2k commented:
> Hi, I tested your PR and the creation of subtask and editing them seems to work as intended. However when i created two subtasks under a task and deleted one of the subtasks the other subtask got deleted too. Here is the video for the same.

@AnishPandita this bug is related to the issue number 12 mentioned in the "Bugs and Functionalities" doc at the "PROJECTS/TASKS/WBS COMPONENT" category. If you reload the page after deleting the task, it will be showed correctly.

vitorsaa2k commented:
Hi @johny4437, Thank you for fixing this bug!

I have just 1 suggestion: 

Add a gap between those buttons: 
![image](https://user-images.githubusercontent.com/17365161/221033250-516b2503-a6f4-4034-8c81-4f5e60a77392.png)

Contact me after making this change, so i can approve this PR ASAP

johny4437 commented:
@vitorsaa2k  Done!
![Captura de Tela (247)](https://user-images.githubusercontent.com/39320057/221369118-87850f49-3660-467b-ba32-a0988606bb25.png)

one-community commented:
Moved to final review team, thank you all

ayush0904 commented:
Hi @johny4437 I have tested the PR and it is working as intended. I have one question. 
I have created a "Parent Task Test" as the parent folder and 2 Child task test as. When I click on the parent folder between the parent and child folder there is another 2 parent folder coming "uijh" and "NewTest_TaskTab" with the sam
<img width="899" alt="Screen Shot 2023-02-25 at 5 50 53 PM" src="https://user-images.githubusercontent.com/15180550/221388066-c8fcb6f3-3d67-4d37-9ea7-fa46549fc26e.png">
e WBS# i.e. 95. Although, this is not an issue but is it possible that the Sub task comes just below the Parent Task?

rauleffting reviewed (changes requested):
Hi, @johny4437! Your PR worked well, as you can see below:

https://user-images.githubusercontent.com/29555732/221584635-c7cd0302-6133-49fb-af10-2757c83489f6.mp4

However, one debug output still remains. Could you delete it? 
Then, I'll approve it to be merged.

Thank you very much!

johny4437 commented on src/components/Projects/WBS/SingleTask/SingleTask.js:
```diff
@@ -13,6 +13,7 @@ import hasPermission from 'utils/permissions';
 
 const SingleTask = props => {
   const taskId = props.match.params.taskId;
+  console.log('TASK ID PARAM', typeof taskId);
```
@rauleffting Done!!

rauleffting reviewed (dismissed):
Hi @johny4437. After your update, the application crashes on the projects page:

Uploading Yo7v7Gh6GlhWvqp.mp4…

I saw that you didn't only delete the debug output, but added code lines. So, maybe that's the cause... 

Thank you very much!

johny4437 commented:
@rauleffting. I will check this!

johny4437 commented:
@rauleffting done! check if the app is running well, and if the functionality of to create a folder is working.

rauleffting reviewed (approved):
Perfect. Johny! And working as intended.

dkamghem commented:
I reviewed this and it works as intended. 
![Screen Shot 2023-03-03 at 3 53 46 PM (2)](https://user-images.githubusercontent.com/83235086/222837926-49bc0ddb-2851-4073-a230-144ab74caccf.png)

one-community commented:
Please don't review already merged PRs Daniel

one-community commented:
Works great, thank you
